libs/vkd3d/state.c

```diff
@@ -3458,6 +3519,69 @@ void vkd3d_fragment_output_pipeline_desc_prepare(struct vkd3d_fragment_output_pi
         desc->dy_info.pDynamicStates = desc->dy_states;
 }
 
+uint32_t vkd3d_fragment_output_pipeline_desc_hash(const void *key)
+{
+    return hash_data(key, sizeof(struct vkd3d_fragment_output_pipeline_desc));
+}
+
+bool vkd3d_fragment_output_pipeline_desc_compare(const void *key, const struct hash_map_entry *entry)
+{
+    const struct vkd3d_fragment_output_pipeline *pipeline = (const void*)entry;
+    const struct vkd3d_fragment_output_pipeline_desc *desc = key;
+    // We zero-initialize these so memcmp is fine
+    return !memcmp(desc, &pipeline->desc, sizeof(*desc));
+}
+
+VkPipeline vkd3d_fragment_output_pipeline_create(struct d3d12_device *device,
+        const struct vkd3d_fragment_output_pipeline_desc *desc)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = &device->vk_procs;
+    struct vkd3d_fragment_output_pipeline_desc desc_copy = *desc;
+    VkGraphicsPipelineLibraryCreateInfoEXT library_create_info;
+    VkGraphicsPipelineCreateInfo create_info;
+    VkPipeline vk_pipeline;
+    VkResult vr;
+
+    vkd3d_fragment_output_pipeline_desc_prepare(&desc_copy);
+
+    memset(&library_create_info, 0, sizeof(library_create_info));
+    library_create_info.sType = VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_LIBRARY_CREATE_INFO_EXT;
+    library_create_info.pNext = &desc_copy.rt_info;
+    library_create_info.flags = VK_GRAPHICS_PIPELINE_LIBRARY_FRAGMENT_OUTPUT_INTERFACE_BIT_EXT;
+
+    memset(&create_info, 0, sizeof(create_info));
+    create_info.sType = VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_CREATE_INFO;
+    create_info.pNext = &library_create_info;
+    create_info.flags = VK_PIPELINE_CREATE_LIBRARY_BIT_KHR | VK_PIPELINE_CREATE_RETAIN_LINK_TIME_OPTIMIZATION_INFO_BIT_EXT;
+    create_info.pColorBlendState = &desc_copy.cb_info;
```

Comment:
Missing pDepthStencil?

Reply:
nvm, was looking the wrong place in spec ...

Reply:
depth-stencil state is part of fragment shader state, not fragment output..